node/src/components/block_proposer.rs

```diff
@@ -275,14 +282,23 @@ impl BlockProposerReady {
     /// Adds a deploy to the block proposer.
     ///
     /// Returns `false` if the deploy has been rejected.
-    fn add_deploy(&mut self, current_instant: Timestamp, hash: DeployHash, header: DeployHeader) {
-        if header.expired(current_instant) {
-            trace!("expired deploy {} rejected from the buffer", hash);
-            return;
+    fn add_deploy_or_transfer(
+        &mut self,
+        current_instant: Timestamp,
+        hash: DeployHash,
+        deploy_or_transfer: DeployType,
+    ) {
+        match deploy_or_transfer {
+            DeployType::Deploy(ref header) | DeployType::Transfer(ref header) => {
+                if header.expired(current_instant) {
```

Comment:
It would make sense to add `expired` to `DeployType`.
Or a `DeployType::header()` method?